composables/api/useLoginUser.ts

```diff
@@ -0,0 +1,57 @@
+import { ref } from 'vue'
+import { useRuntimeConfig } from '#app'
+import { useCurrentUser, useSessionToken } from '@/composables'
+import { LoginRequest, ProblemDetails } from '@/lib/api/data-contracts'
+import { Users } from '@/lib/api/Users'
+
+interface LoginUserResponse {
+  loading: boolean
+  error: ProblemDetails | null
+}
+
+export const useLoginUser = (requestData: LoginRequest): LoginUserResponse => {
+  const responseLoading = ref(true)
+  const responseError = ref<ProblemDetails | null>(null)
+  const authToken = useSessionToken()
+  const currentUser = useCurrentUser()
+
+  const userClient = new Users({
+    baseUrl: useRuntimeConfig().public.BACKEND_BASE_URL,
+  })
+  userClient
+    .usersLoginCreate(requestData)
+    .then((response) => {
```

Comment:
is it possible to rewrite this into `async / await` syntax? If not that's fine but just to keep it as concise as possible